src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -6,295 +6,192 @@
-              (int) (points[numTerritories - 1].y + textyOffset - yOffset + translateY), null);
-          graphics.drawImage(movementImage, (int) (points[numTerritories - 1].x + textXOffset - xOffset),
-              (int) (points[numTerritories - 1].y + textyOffset - yOffset - translateY), null);
-        }
-        if (scrollWrapX && scrollWrapY) {
-          graphics.drawImage(movementImage, (int) (points[numTerritories - 1].x + textXOffset - xOffset + translateX),
-              (int) (points[numTerritories - 1].y + textyOffset - yOffset + translateY), null);
-          graphics.drawImage(movementImage, (int) (points[numTerritories - 1].x + textXOffset - xOffset - translateX),
-              (int) (points[numTerritories - 1].y + textyOffset - yOffset - translateY), null);
-        }
-        final Image cursorImage = routeDescription.getCursorImage();
-        if (cursorImage != null) {
-          graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset),
-              (int) (points[numTerritories - 1].y + cursorYOffset - yOffset), null);
-          if (scrollWrapX /* && !scrollWrapY */) {
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset + translateX),
-                (int) (points[numTerritories - 1].y + cursorYOffset - yOffset), null);
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset - translateX),
-                (int) (points[numTerritories - 1].y + cursorYOffset - yOffset), null);
-          }
-          if (/* !scrollWrapX && */scrollWrapY) {
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset),
-                (int) (points[numTerritories - 1].y + cursorYOffset - yOffset + translateY), null);
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset),
-                (int) (points[numTerritories - 1].y + cursorYOffset - yOffset - translateY), null);
-          }
-          if (scrollWrapX && scrollWrapY) {
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset + translateX),
-                (int) (points[numTerritories - 1].y + cursorXOffset - yOffset + translateY), null);
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset - translateX),
-                (int) (points[numTerritories - 1].y + cursorXOffset - yOffset - translateY), null);
-          }
-        }
-      }
-    } finally {
-      graphics.setTransform(original);
+    // set thickness and color of the future drawings
+    graphics.setStroke(new BasicStroke(3.5f, BasicStroke.CAP_ROUND, BasicStroke.JOIN_ROUND));
+    graphics.setPaint(Color.red);
+    graphics.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);
+    if (routeDescription == null) {
```

Comment:
does not seem like graphics should be changed unless the method will actually do something. The check for null and return statements should probably be before we do anything else.